salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
                     raise CommandExecutionError('Unable to write mount cache.')
     return True
+
+
+def _filesystems(config='/etc/filesystems', leading_key=True):
+    '''
+    .. versionadded:: 2018.3.3
+
+    List the contents of the filesystems
+
+    config
+        File containing filesystem infomation
+
+    leading_key
+        True    return dictionary keyed by 'name' value and dictionary with other keys, values
+                { '/dir' : { 'dev': '/dev/hd8', .... }}
+
+        False   return dictionary  keyed by 'name' value and dictionary with all keys, values
+                { '/dir' : { 'name' : '/dir', 'dev': '/dev/hd8', .... }}
+    '''
+    ret = OrderedDict()
+    lines = []
+    parsing_block = False
+    if not os.path.isfile(config) or 'AIX' not in __grains__['kernel']:
+        return ret
+
+    # read in block of filesystems, block starts with '/' till empty line
+    with salt.utils.files.fopen(config) as ifile:
+        for line in ifile:
+            line = salt.utils.stringutils.to_unicode(line)
+
+            # skip till first entry
+            if not line.startswith('/') and not parsing_block:
+                continue
+
+            if line.startswith('/'):
+                parsing_block = True
+                lines.append(line)
+            elif not line.split():
+                parsing_block = False
+                try:
+                    entry = _filesystems_entry.dict_from_lines(
+                        lines,
+                        _filesystems_entry.compatibility_keys)
+                    lines = []
+                    if 'opts' in entry:
+                        entry['opts'] = entry['opts'].split(',')
+                    while entry['name'] in ret:
+                        entry['name'] += '_'
+
+                    if leading_key:
+                        ret[entry.pop('name')] = entry
+                    else:
+                        ret[entry['name']] = entry
+
+                except _filesystems_entry.ParseError:
+                    pass
+            else:
+                lines.append(line)
+
+    return ret
+
+
+def filesystems(config='/etc/filesystems'):
+    '''
+    .. versionadded:: 2018.3.3
+
+    List the contents of the filesystems
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mount.filesystems
+    '''
+    ret = {}
+    if 'AIX' not in __grains__['kernel']:
+        return ret
+
+    ret_dict = _filesystems(config)
+    if ret_dict:
+        ret_key = ret_dict.keys()[0]
+        ret = {ret_key: dict(ret_dict[ret_key])}
+
+    return ret
+
+
+def set_filesystems(
+        name,
+        device,
+        vfstype,
+        opts='-',
+        mount='true',
+        config='/etc/filesystems',
+        test=False,
+        match_on='auto',
+        **kwargs):
+    '''
+    .. versionadded:: 2018.3.3
+
+    Verify that this mount is represented in the filesystems, change the mount
+    to match the data passed, or add the mount if it is not present on AIX
```

Comment:
I'm not sure how I follow that the functionality of this is different than the `remount` function

Reply:
@cachedout so what is the reason for set_automaster, set_fstab, set_vfstab, just being consistent with existing functionality which is used internally in states/mount.py, where set_fstab is called.  Need to have AIX equivalents hence set_filesystems.